### PR TITLE
Fix for 1553

### PR DESCRIPTION
**Pull request by sbrossie**
This matches the api changes introduced in this [PR](https://github.com/killbill/killbill-api/pull/111).

I broke the set of changes in 3 phases (might be easier for review):
* Core implementation to replace the `invoice_payments#success` ->  `invoice_payments#status`: https://github.com/killbill/killbill/pull/1802/commits/9f56e9344837c806a6122c02af08c8f7ef326c7d https://github.com/killbill/killbill/pull/1802/commits/58f8cb36592ece9603240e4b30107a948b66f0c9
* Modify InvoiceInternalApi to pass the InvoicePaymentStatus:  https://github.com/killbill/killbill/pull/1802/commits/613943d454d44e77e819eab3187b2506b359aac3
* Fix for #1553: https://github.com/killbill/killbill/pull/1802/commits/246bbda3e6bb00251a0813eb195b67ab38c8ff65

Also see:
* This [comment](https://github.com/killbill/killbill-api/pull/111#issue-1458980872) with regards to the new states introduced.
* The `invoicePayment#state` is not visible at the JAXRS layer (hence no client to change). This is because there is a merge happening between Payments and InvoicePayments so this does not seem needed.

### CODE REVIEW

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestInvoicePayment.java [hunk 1]
```diff
@@ -1172,7 +1187,7 @@ public void testWithPendingPaymentThenFailure() throws Exception {
         assertEquals(invoice1.getPayments().size(), 1);
         assertEquals(invoice1.getPayments().get(0).getAmount().compareTo(new BigDecimal("249.95")), 0);
         assertEquals(invoice1.getPayments().get(0).getCurrency(), Currency.USD);
-        assertFalse(invoice1.getPayments().get(0).isSuccess());
+        assertFalse(invoice1.getPayments().get(0).getStatus() == InvoicePaymentStatus.SUCCESS);
```
Comment (pierre) [inline beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestInvoicePayment.java:1190]: Maybe be explicit and assert the right state here (vs `assertFalse`)?

File: invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java [hunk 2]
```diff
@@ -836,7 +837,7 @@ public InvoicePaymentModelDao postChargebackReversal(final UUID paymentId, final
                                         invoicePayment.getProcessedCurrency(),
                                         invoicePayment.getPaymentCookieId(),
                                         invoicePayment.getLinkedInvoicePaymentId() == null ? null : invoicePayment.getLinkedInvoicePaymentId().toString(),
-                                        false,
+                                        InvoicePaymentStatus.INIT.toString(),
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java:840]: This is where a fourth state could have been useful: when the chargeback is reversed, it's not really _INIT_ (and it's not really _FAILED_ either, see https://github.com/killbill/killbill/issues/477#issuecomment-220989853).

This is such a rare case though, not worth optimizing.

File: payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java [hunk 3]
```diff
@@ -179,9 +180,8 @@ public OnSuccessPaymentControlResult onSuccessCall(final PaymentControlContext p
                             invoicePaymentAmount = paymentControlContext.getAmount();
                         }
 
-                        log.debug("Notifying invoice of {} paymentId='{}', amount='{}', currency='{}', invoiceId='{}'", success ? "successful" : "pending", paymentControlContext.getPaymentId(), invoicePaymentAmount, paymentControlContext.getCurrency(), invoiceId);
+                        log.debug("Notifying invoice of {} paymentId='{}', amount='{}', currency='{}', invoiceId='{}' invoicePaymentStatus='{}'", status, paymentControlContext.getPaymentId(), invoicePaymentAmount, paymentControlContext.getCurrency(), invoiceId);
```
Comment (pierre) [inline payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java:183]: ⚠️ Wrong ordering / numbering of arguments.

**Review by pierre — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -61,7 +61,7 @@
     <properties>
         <check.skip-dependency-versions>true</check.skip-dependency-versions>
         <check.spotbugs-exclude-filter-file>${main.basedir}/spotbugs-exclude.xml</check.spotbugs-exclude-filter-file>
-        <killbill-api.version>0.54.0-9b20b32-SNAPSHOT</killbill-api.version>
+        <killbill-api.version>0.54.0-7a22bc7-SNAPSHOT</killbill-api.version>
         <killbill-client.version>1.3.0-a6e0394-SNAPSHOT</killbill-client.version>
```
Comment (reshmabidikar) [inline pom.xml:65]: Could you please specify the `killbill-api` version as `0.54.0-e27330f-SNAPSHOT` (latest version of `killbill-api`)?

**Review by reshmabidikar — APPROVED** [review]
I did not go into too much detail as I do not understand the issue completely but looks good overall.

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestInvoicePayment.java (hunk 1 above)
Comment (sbrossie) [reply beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestInvoicePayment.java]: Fixed in https://github.com/killbill/killbill/pull/1802/commits/22883101537984447dbfd4b5fc26b183ac3a2d4c

File: payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java (hunk 3 above)
Comment (sbrossie) [reply payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java]: Fixed in https://github.com/killbill/killbill/pull/1802/commits/580bee255fc583ae6ca799ff622cc1beede1ac75

File: pom.xml (hunk 4 above)
Comment (sbrossie) [reply pom.xml]: I merged your PR and as part of the merge got those versions updated.

**Pull request merged**